docs/orchestration/tutorial/first.md

```diff
@@ -37,6 +37,23 @@ flow.register(project_name="Hello, World!")
 Registration only sends data about the existence and format of your flow; **no actual code from the flow is sent to the Prefect API**. Your code remains safe, secure, and private in your own infrastructure!
 :::
 
+:::tip Deduplicating registration calls
+Each call to `flow.register()` will bump the version of the flow in the backend.
+If you are registering flows using automation, you may want to pass an `idempotency_key` which will only create a new version when the key changes.
+For example:
+```python
+MY_INTERNAL_FLOW_VERSION = "1.0.1"
+# This constant needs to be manually bumped for re-registration
+# and could instead be generated from the last commit to the
+# flow file or another indicator
```

Comment:
this might be overly complicated to suggest in our docs, but a consistent hash of `flow.serialize()` also works, something like:
```python
hashlib.sha256(json.dumps(flow.serialize()).encode()).hexdigest()
```
ultimately, if the JSON payload is identical between registrations then Cloud is gaining no new information.